documentation/src/main/asciidoc/topics/ref_discovery_cloud.adoc

```diff
@@ -0,0 +1,54 @@
+[id='jgroups-cloud-discovery-protocols_{context}']
+= Cloud Discovery Protocols
+
+{brandname} includes default JGroups stacks that use discovery protocol implementations that are specific to cloud providers.
+
+[%header,cols=4*]
+|===
+|Discovery protocol
+|Default stack file
+|Artifact
+|Version
+
+|`NATIVE_S3_PING`
+|`default-jgroups-ec2.xml`
+|`org.jgroups.aws.s3:native-s3-ping`
+|`0.9.5`
+
+|`GOOGLE_PING2`
+|`default-jgroups-google.xml`
+|`org.jgroups.google:jgroups-google`
+|`1.0.0`
+
+|`AZURE_PING`
+|`default-jgroups-azure.xml`
+|`org.jgroups.azure:jgroups-azure`
+|`1.2.1.Final`
+|===
+
+[discrete]
+== Providing Dependencies for Cloud Discovery Protocols
+
+To use `NATIVE_S3_PING`, `GOOGLE_PING2`, or `AZURE_PING` cloud discovery protocols, you need to provide dependent libraries to {brandname}.
+
+.Procedure
+
+ifdef::embedded_caches[]
+* Add the artifact dependencies to your project `pom.xml`.
+endif::embedded_caches[]
+
+ifdef::remote_caches[]
+. Download the artifact JAR file from the Maven Central repository.
+. Add the artifact JAR file to the `{server_home}/server/lib` directory of your {brandname} Server installation.
```

Comment:
you need the artifact jar and its dependencies. For example, for native-s3 you will need
```
org.jgroups.aws.s3:native-s3-ping:1.0.0.Final
com.amazonaws:aws-java-sdk-core:1.11.708
commons-logging:commons-logging:1.1.3
org.apache.httpcomponents:httpclient:4.5.9
org.apache.httpcomponents:httpcore:4.4.11
commons-codec:commons-codec:1.11
software.amazon.ion:ion-java:1.0.2
joda-time:joda-time:2.8.1
com.amazonaws:aws-java-sdk-s3:1.11.708
com.amazonaws:aws-java-sdk-kms:1.11.708
com.amazonaws:jmespath-java:1.11.708
```

Tip you can use `mvn dependency:copy -Dartifact=com.amazonaws:aws-java-sdk-core:1.11.708 -DoutputDirectory=/path/to/server/lib` to donwload from maven central

Reply:
@pruivo Nice one. I adapted this for the native s3 ping artifact. Hopefully the changes are sensible.

Reply:
@pruivo I've removed the details about mvn usage as we talked about on zulip. I'll work on the KCS and we can add a link to this topic.